src/main/java/duke/Task.java

```diff
@@ -0,0 +1,48 @@
+package duke;
+import java.time.LocalDate;
+
+/**
+ * The class in which Deadline, Event and ToDo inherits from.
+ * It contains several common methods shared by all subclasses.
+ */
+public abstract class Task {
+    public final String taskName;
+    public boolean isDone;
+
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    public Task(String taskName, boolean isDone) {
+        this.taskName = taskName;
+        this.isDone = isDone;
+    }
+
+    public void completeTask() {
+        this.isDone = true;
+    }
+
+    public String getTaskName() {
+        return this.taskName;
+    }
+
+    /**
+     * Takes in a LocalDate object and returns the date of the object in a special format.
+     *
+     * @param date the LocalDate object to print the date of.
+     * @return the string representation of the date of the object in the special format.
+     */
+    public static String printDate(LocalDate date) {
```

Comment:
I think you can use the inbuilt methods to achieve your desired results. [https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDate.html#format(java.time.format.DateTimeFormatter)](url)